src/main/java/fr/minuskube/inv/SmartInvsPlugin.java

```diff
@@ -16,34 +16,61 @@
 
 package fr.minuskube.inv;
 
+import org.bukkit.plugin.Plugin;
 import org.bukkit.plugin.java.JavaPlugin;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
 
-public class SmartInvsPlugin extends JavaPlugin {
+/**
+ * Main plugin class of SmartInventory plugin/library.
+ *
+ * @since 1.0
+ */
+public final class SmartInvsPlugin extends JavaPlugin {
 
-    private static JavaPlugin instance;
-    private static InventoryManager invManager;
+    @Nullable
+    private static Plugin instance;
 
-    @Override
-    public void onEnable() {
-        setPlugin(this);
+    @Nullable
+    private static InventoryManager manager;
+
+    @NotNull
+    public static InventoryManager getManager() {
+        if (SmartInvsPlugin.manager == null) {
+            throw new IllegalStateException("You can't use SmartInvsPlugin#getInstance() method before plugin start!");
+        }
+        return SmartInvsPlugin.manager;
+    }
+
+    @NotNull
+    public static Plugin getInstance() {
+        if (SmartInvsPlugin.instance == null) {
+            throw new IllegalStateException("You can't use SmartInvsPlugin#getInstance() method before plugin start!");
+        }
+        return SmartInvsPlugin.instance;
     }
 
     @Override
     public void onDisable() {
-        deleteStaticReferences();
+        super.onDisable();
+        SmartInvsPlugin.deleteStaticReferences();
     }
 
-    public static InventoryManager manager() { return invManager; }
-    public static JavaPlugin instance() { return instance; }
+    private static void deleteStaticReferences() {
+        SmartInvsPlugin.instance = null;
+        SmartInvsPlugin.manager = null;
+    }
 
-    public static void setPlugin(JavaPlugin javaPlugin) {
-        instance = javaPlugin;
-        invManager = new InventoryManager(javaPlugin);
-        invManager.init();
+    @Override
+    public void onEnable() {
+        super.onEnable();
```

Comment:
Unnecessary call to that method, as it is just an empty method body in the JavaPlugin class